### ISSUE TITLE
findOne( { where: { post, user } }) not compiling with TypeScript

### ISSUE THREAD

**Issue by AaronNGray** [typeorm/typeorm#9794]
### Issue description

non unary 'where's are now not working

### Expected Behavior

I was using typeorm@0.2.41 and 
```
const postLikes:PostLikes | null = await PostLikes.findOne({ post, user});
```
just worked !

Now with typeorm@latest do you realize how much code you have broken ???
I have to change every occurance, but non unary 'where's are now not working !
```
const postLikes:PostLikes | null = await PostLikes.findOne({ where: {post, user} });
```
this should still should compile ?

### Actual Behavior

is not compiling :-
```
src/resolvers/Post.ts:52:74 - error TS2322: Type '{ post: Post; user: User; }' is not assignable to type 'FindOptionsWhere<PostLikes> | FindOptionsWhere<PostLikes>[] | undefined'.
  Types of property 'post' are incompatible.
    Type 'Post' is not assignable to type 'boolean | FindOperator<any> | FindOptionsWhere<Post> | FindOptionsWhere<Post>[] | EqualOperator<Post> | undefined'.
      Type 'Post' is not assignable to type 'FindOptionsWhere<Post>'.
        Types of property 'user' are incompatible.
          Type 'User' is not assignable to type 'boolean | FindOperator<any> | FindOptionsWhere<User> | FindOptionsWhere<User>[] | EqualOperator<User> | undefined'.
            Type 'User' is not assignable to type 'FindOptionsWhere<User>'.
              Types of property 'posts' are incompatible.
                Type 'Post[]' is not assignable to type 'boolean | FindOperator<any> | FindOptionsWhere<Post> | FindOptionsWhere<Post>[] | EqualOperator<Post> | undefined'.

52             const postLikes:PostLikes | null = await PostLikes.findOne({ where: {post, user} });
```

### Steps to reproduce

mn, will do when I get time !

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |  Windows 10 |
| Node.js version     | 18.12.1 |
| Typescript version  | 4.9.5 |
| TypeORM version     | 0.3.12 |

### Additional Context

Do you realize how much code you have broken and people will not want to upgrade their code bases and there may well be many many unmaintained code bases left behind.

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [X] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [ ] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

Yes, I have the time, but I don't know how to start. I would need guidance.

**Comment by psecuresystem** [comment]
I think the code should be:
```js
const postLikes:PostLikes | null = await PostLikes.findOne({ where: {post_id: post.id, user_id: user.id} });
```
Obviously this is assuming that you have a post and user already in the context.
Again I'm not sure but try it.

**Comment by AaronNGray** [comment]
@psecuresystem no that does not seem to work, thanks anyway.

**Comment by lawrentiy** [comment]
The same issue after update from 
"typeorm": "^0.2.41" 
up to
"typeorm": "^0.3.*"

**Comment by lawrentiy** [comment]
> The same issue after update from "typeorm": "^0.2.41" up to "typeorm": "^0.3.*"

Have to downgrade unfortunately((

**Comment by AaronNGray** [comment]
@lawrentiy - I am delaying updating until this issue is fixed.

**Comment by kattae23** [comment]
Something new? i have the same issue

**Comment by kattae23** [comment]
i resolve it, 

`const userCourses = await this.userCourseRepository.find({
      where: {
        user: {
          id: userId,
        },
      },
      loadRelationIds: true,
    });`
    
    🤓🤓🤓🤓

**Comment by AaronNGray** [comment]
@kattae23 - Thanks Alphonso, just the `loadRelationIds: true` flag.

Did you find this in documentation or in code :-
https://github.com/typeorm/typeorm/blob/022d2b5f622771349355f00a087c26c930db0d25/src/query-builder/SelectQueryBuilder.ts#L3320

**Comment by AaronNGray** [comment]
I will close the issue once I have tested it in my code.

**Comment by kattae23** [comment]
@AaronNGray is not only the loadRelationIds, you need to open an object in the schema and select the id inside it

**Comment by kattae23** [comment]
This is your case:
``const postLikes:PostLikes | null = await PostLikes.findOne({ post: {
   id: post.id
}, user : {
   id: user.id
}
});

**Comment by AaronNGray** [comment]
Yes that does not work, I will try it with the `loadRelationIds: true` predicate.
